### DISCUSSION TITLE
Can I describe a StoreState as an object with propertis?

### DISCUSSION THREAD

**Discussion by IvayloGugalov** [pmndrs/zustand#1744] [Q&A] [answered]
Hi, I was looking at the [Typescript guide](https://github.com/pmndrs/zustand/blob/main/docs/guides/typescript.md) and I was wondering can multiple slices be implemented like this:
```ts
interface BearSlice {
  bears: number
}

const createBearSlice: StateCreator<BearSlice> = (set) => ({
  bears: 0
})

interface FishSlice {
  fishes: number
}

interface FishSlice {
  fishes: number
}
const createFishSlice: StateCreator<FishSlice> = (set) => ({
  fishes: 0
})

interface StoreState {
  bears: BearSlice,
  fishes: FishSlice
}

const useBoundStore = create<StoreState>()((...a) => ({
  bears: createBearSlice(...a),
  fishes: createFishSlice(...a),
}))
```
As I am not able to make this work I was wondering if it's even possible or if it's better to use `BearSlice & FishSlice`

**Comment by dbritto-dev** [comment]
Hi @IvayloGugalov, what you are looking for is the slices pattern. Here's the guide https://docs.pmnd.rs/zustand/guides/slices-pattern

The solution shall looks like this:

```ts
const useBoundStore = create<StoreState>()((...a) => ({ 
  ...createBearSlice(...a),
  ...createFishSlice(...a), 
}))
```

Having `bears: createBearSlice(...a)` returns `bears: { bears: 0 }` same for `fishes: createFishSlice(...a)`

About the `StoreState` the solution shall looks like this:


```ts
type StoreState = BearSlice & FishSlice
```

**Comment by IvayloGugalov** [comment]
My goal is that when I call  `useStore()` in my component to deconstruct it as
```const { bearsStore } = useStore();``` and use it as ```bearsStore.bears```
With the given approach the code is decosnstructing the store with all the slices merged into a single object.
And when I try to build the StoreState as
```ts
type StoreState = { bearStore: BearSlice, fishStore: FishSlice };
export const initializeStore = () => {
  return create<StoreState>((...a) => ({
    bearStore: createBearSlice(...a),
    fishStore: createFishSlice(...a)
  }))
}
``` 
I get the issue that I am passing the wrong type of StateCreator to each slice and I was hoping that there is a way of passing the correct type of StateCreator.

**Reply by dbritto-dev** [reply]
In that case would be better to have multiple stores instead of trying to have them using slices. I would use `useBearStore()` and `useFisheStore()` rather than `useStore()`

**Reply by dbritto-dev** [reply, marked as answer]
BTW, what you are looking for is something like this: https://codesandbox.io/p/sandbox/crimson-architecture-jry04s?file=README.md but that is wrong since you are mixing the slices and multiple stores which currently is not possible without hacking types. So my suggestions is to use multiple stores and not mixing them with slices.

**Reply by dbritto-dev** [reply]
@IvayloGugalov ping me if you have any questions

**Reply by IvayloGugalov** [reply]
@dbritto-dev thank you for your time and demo code. Will stick with multiple stores.
